packages/smart-contracts/packages/nft/contracts/Nft.sol

```diff
@@ -228,6 +228,9 @@ contract Nft is
         // Effects: deactivate the old NFT
         s_nftData[oldTokenId].activityType = ActivityType.DEACTIVATED;
 
+        // Effects: Update the active NFT for the sender
+        delete s_activeNft[receiver];
+
         emit NftDeactivated(oldTokenId);
```

Comment:
let's have this emit instantly after deactivation and not deletion action (move to 230 line with spacing)